SConstruct

```diff
@@ -507,21 +510,11 @@ if selected_platform in platform_list:
         env.extra_suffix += "." + env["extra_suffix"]
 
     # Environment flags
-    CCFLAGS = env.get("CCFLAGS", "")
-    env["CCFLAGS"] = ""
-    env.Append(CCFLAGS=str(CCFLAGS).split())
-
-    CFLAGS = env.get("CFLAGS", "")
-    env["CFLAGS"] = ""
-    env.Append(CFLAGS=str(CFLAGS).split())
-
-    CXXFLAGS = env.get("CXXFLAGS", "")
-    env["CXXFLAGS"] = ""
-    env.Append(CXXFLAGS=str(CXXFLAGS).split())
-
-    LINKFLAGS = env.get("LINKFLAGS", "")
-    env["LINKFLAGS"] = ""
-    env.Append(LINKFLAGS=str(LINKFLAGS).split())
+    env.Append(CPPDEFINES=env.get("cppdefines", ""))
```

Comment:
Thanks, I was afraid of that.

I'm worried that `env.get(...).split()` might sometimes fails because `env.get()` doesn't necessarily return a string. Notably in recent SCons version, I've seen it sometimes we a `deque`... and thus even the previous `str(...).split()` wouldn't work because stringification of `deque` seems to do something like `deque(...)`.

My Python-fu isn't great there, but we'd need to find something that we're confident will work for both strings and deques I guess.

Reply:
Ah well I guess it was a `deque` when we were getting `env["CPPDEFINES"]`. For a command line argument that's not conflicting with a SCons Environment collection, I guess we might be good assuming that we're dealing with a string?

```diff
diff --git a/SConstruct b/SConstruct
index e51719b3f8..777faa3fd7 100644
--- a/SConstruct
+++ b/SConstruct
@@ -510,6 +510,8 @@ if selected_platform in platform_list:
         env.extra_suffix += "." + env["extra_suffix"]
 
     # Environment flags
+    print(type(env.get("cppdefines")))
+    print(type(env["CPPDEFINES"]))
     env.Append(CPPDEFINES=env.get("cppdefines", ""))
     env.Append(CCFLAGS=env.get("ccflags", ""))
     env.Append(CXXFLAGS=env.get("cxxflags", ""))
```

```
$ scons cppdefines="test test1"
<class 'str'>
<class 'collections.deque'>
```